src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_params.py

```diff
@@ -26,37 +24,60 @@
+        c.argument('storage_account_name', help='Provide the name of an existing storage account if you\'re recreating a container registry over a previous registry created storage account. Only applicable to Classic SKU.', completer=get_resource_name_completion_list(STORAGE_RESOURCE_TYPE))
+        c.argument('sku', help='The SKU of the container registry', choices=MANAGED_REGISTRY_SKU + CLASSIC_REGISTRY_SKU)
+        c.argument('password_name', help='The name of password to regenerate', choices=['password', 'password2'])
+        c.argument('username', options_list=('--username', '-u'), help='The username used to log into a container registry')
+        c.argument('password', options_list=('--password', '-p'), help='The password used to log into a container registry')
+
+    with self.argument_context('acr create') as c:
+        c.argument('admin_enabled', nargs='?', required=False, const='true', default='false', help='Indicates whether the admin user is enabled', choices=['true', 'false'])
+
+    with self.argument_context('acr update') as c:
+        c.argument('admin_enabled', help='Indicates whether the admin user is enabled', choices=['true', 'false'])
+
+    with self.argument_context('acr repository delete') as c:
+        c.argument('manifest', nargs='?', required=False, const='', default=None, help='The sha256 based digest of manifest to delete')
+        c.argument('yes', options_list=('--yes', '-y'), action='store_true', help='Do not prompt for confirmation')
+        c.argument('repository', help='The name of the repository to delete.')
+        c.argument('tag', help='The name of tag to delete')
+
+    with self.argument_context('acr repository show-manifests') as c:
+        c.argument('repository', help='The repository to obtain manifests from.')
+
+    with self.argument_context('acr repository show-tags') as c:
+        c.argument('repository', help='The repository to obtain tags from.')
+
+    with self.argument_context('acr create') as c:
+        c.argument('registry_name', completer=None, validator=validate_registry_name)
+        c.argument('deployment_name', arg_type=deployment_name_type, validator=None)
+        c.argument('location', arg_type=get_location_type(self.cli_ctx), validator=get_default_location_from_resource_group)
+
+    with self.argument_context('acr check-name') as c:
+        c.argument('registry_name', completer=None)
+
+    with self.argument_context('acr webhook') as c:
+        c.argument('registry_name', options_list=('--registry', '-r'), help='The name of the container registry. You can configure the default registry name using `az configure --defaults acr=<registry name>`', completer=get_resource_name_completion_list(REGISTRY_RESOURCE_TYPE), configured_default='acr')
+        c.argument('webhook_name', options_list=('--name', '-n'), help='The name of the webhook', completer=get_resource_name_completion_list(WEBHOOK_RESOURCE_TYPE))
```

Comment:
Yugang has a magic regex to turn all of these option_list tuples into lists.  Could you run that on this file?

Reply:
Done.
Out of interest, is there a difference between the two or it's preference?

Reply:
The tuple format is annoyingly restrictive. If you wanted just to change the long-option you have to put `(--foo,)` and if you missed the comma then you'd spend a bunch of time investigating cryptic errors. Hence the list format was added so you can specify an arbitrary number of aliases. I'd like to excise them so newcomers don't see the tuple approach and think it's the best way.